src/metatracer.rs

```diff
@@ -36,11 +36,13 @@ pub type HotThreshold = u32;
 const DEFAULT_HOT_THRESHOLD: HotThreshold = 50;
 
 // The current meta-tracing phase of a given location in the end-user's code. Consists of a tag and
-// (optionally) a value.
-const PHASE_TAG     : u32 = 0b11; // All of the other PHASE_ tags must fit in this.
-const PHASE_COUNTING: u32 = 0b00; // The value specifies the current hot count.
-const PHASE_TRACING : u32 = 0b01;
-const PHASE_COMPILED: u32 = 0b10;
+// (optionally) a value. The tags are in the high order bits since we expect the most common tag is
+// PHASE_COMPILED which (one day) will have an index associated with it: tagging the top-most bits
+// means that we can access the index simply with "&" (with no shifting required).
```

Comment:
This is a good point, that we expect the "compiled" state to be most frequent. That being the case, why don't you make `PHASE_COMPILED` the `0b00` case? Then you won't have to mask away tags to get the index since the top bits are already `00`?

Reply:
Good idea.

There is already a test which checks the incrementing BTW.

Reply:
Right, so my next question is, why didn't it spot the bogus increment cause by the over-zealous shift?

Reply:
All that was wrong was that we shifted everything (left and right) 1 bit too many. The incrementing was fine. Basically we accidentally wasted one bit of storage, but otherwise there were no harmful effects.

Reply:
Are you sure? As I see it, you lost the bottom bit of the existing counter value (thus subtracting one if the counter was an odd value), and then incremented the next bit up, thus in effect incrementing the counter by 2.

Reply:
Your best bet is to try damaging the code and see if the test fails.

[It does FWIW, but I don't think I'm going to convince you until you try!]

Reply:
Ugh, I see, since you interpret `1<<2` as `1<<1` it works.

All right, but the principal is the same. If I were to replace the increment with a double increment, would a test fail? I don't think it would.

Reply:
Why not try it and see what happens?

Reply:
[I've tried it but, as I said, you won't believe me until you've tried it ;) ]

Reply:
> [It does FWIW, but I don't think I'm going to convince you until you try!]

It doesn't work if both of us use Socratic questioning :P

Reply:
You've out-Classic'd me. I don't know what that means. I do know that I was right that you wouldn't believe me ;) Let me know when you've tried this out and convinced yourself (or found a bug... I might be wrong!).

Reply:
Actually your tests are more clever than I thought.

And, I've just spent about half an hour deciding if you are right to use relaxed orderings here. I think you are. (head hurts).